web/modules/custom/joinup_core/src/Plugin/Validation/Constraint/OptionalRequiredLocationFieldsConstraint.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Drupal\joinup_core\Plugin\Validation\Constraint;
+
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * Checks if one of the given fields are empty or not.
+ *
+ * @Constraint(
+ *   id = "OptionalRequiredLocationFields",
+ *   label = @Translation("Optional required location fields constraint", context = "Validation"),
+ * )
+ */
+class OptionalRequiredLocationFieldsConstraint extends Constraint {
```

Comment:
The original idea was that make it general is as possible but since we only want to use in limited cases then it should be moved to the joinup_event module so you're right!